diff --git a/configure.ac b/configure.ac
index 9a286c2..aa95d3f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -297,21 +297,21 @@ dnl -Wundef: too many broken headers
 AG_GST_SET_ERROR_CFLAGS($FATAL_WARNINGS, [
       -Wmissing-declarations -Wmissing-prototypes -Wredundant-decls
       -Wwrite-strings -Wold-style-definition -Waggregate-return
-      -Winit-self -Wmissing-include-dirs -Waddress -Wno-multichar
+      -Winit-self -Wnomissing-include-dirs -Waddress -Wno-multichar
       -Wnested-externs $NO_WARNINGS])
 
 dnl define an ERROR_CXXFLAGS Makefile variable
 AG_GST_SET_ERROR_CXXFLAGS($FATAL_WARNINGS, [
         -Wmissing-declarations -Wredundant-decls
         -Wwrite-strings
-        -Winit-self -Wmissing-include-dirs -Waddress -Wno-multichar
+        -Winit-self -Wnomissing-include-dirs -Waddress -Wno-multichar
         $NO_WARNINGS])
 
 dnl define an ERROR_OBJCFLAGS Makefile variable
 AG_GST_SET_ERROR_OBJCFLAGS($FATAL_WARNINGS, [
       -Wmissing-declarations -Wmissing-prototypes -Wredundant-decls
       -Wwrite-strings -Wold-style-definition
-      -Winit-self -Wmissing-include-dirs -Wno-multichar
+      -Winit-self -Wnomissing-include-dirs -Wno-multichar
       -Wnested-externs $NO_WARNINGS])
 
 dnl define correct level for debugging messages
