diff --git a/scipy/special/setup.py b/scipy/special/setup.py
index 8f74f0502ab7..06938819a898 100755
--- a/scipy/special/setup.py
+++ b/scipy/special/setup.py
@@ -80,10 +80,12 @@ def configuration(parent_package='',top_path=None):
     cfg.setdefault('libraries', []).extend(['sc_amos','sc_c_misc','sc_cephes','sc_mach',
                                             'sc_cdf', 'sc_specfun'])
     cfg.setdefault('define_macros', []).extend(define_macros)
+    extra_info = get_info("npymath")
+    extra_info['library_dirs'] = ["=/usr/lib/python3.5/site-packages/numpy/core/lib"]
     config.add_extension('_ufuncs',
                          depends=ufuncs_dep,
                          sources=ufuncs_src,
-                         extra_info=get_info("npymath"),
+                         extra_info=extra_info,
                          **cfg)
 
     # Extension _ufuncs_cxx
@@ -96,7 +98,7 @@ def configuration(parent_package='',top_path=None):
                          depends=ufuncs_cxx_dep,
                          include_dirs=[curdir],
                          define_macros=define_macros,
-                         extra_info=get_info("npymath"))
+                         extra_info=extra_info)
 
     cfg = dict(get_system_info('lapack_opt'))
     config.add_extension('_ellip_harm_2',
diff --git a/scipy/spatial/setup.py b/scipy/spatial/setup.py
index 9492dd1caea7..5499f2069724 100755
--- a/scipy/spatial/setup.py
+++ b/scipy/spatial/setup.py
@@ -72,12 +72,14 @@ def configuration(parent_package='', top_path=None):
                          sources=['ckdtree.cxx'] + ckdtree_src,
                          depends=ckdtree_dep,
                          include_dirs=inc_dirs + [join('ckdtree','src')])
+    extra_info = get_misc_info("npymath")
+    extra_info['library_dirs'] = ["=/usr/lib/python3.5/site-packages/numpy/core/lib"]
     # _distance_wrap
     config.add_extension('_distance_wrap',
         sources=[join('src', 'distance_wrap.c')],
         depends=[join('src', 'distance_impl.h')],
         include_dirs=[get_numpy_include_dirs()],
-        extra_info=get_misc_info("npymath"))
+        extra_info=extra_info)
 
     return config
 
