diff --git a/configure.ac b/configure.ac
index 13df2637868c..d8d2a189a4d8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -208,15 +208,15 @@ if test "x$qt_pkgconfig" = "xtrue"; then
 else
    PKG_CHECK_MODULES(QT, [QtCore >= 4.0 QtGui >= 4.0], [qt_pkgconfig=true], [qt_pkgconfig=false])
    if test "x$qt_pkgconfig" = "xtrue"; then
-      MOC=`$PKG_CONFIG --variable=moc_location QtCore`
-      UIC=`$PKG_CONFIG --variable=uic_location QtCore`
-      RCC=`$PKG_CONFIG --variable=rcc_location QtCore`
-      if test -z "$RCC"; then
-         RCC="rcc"
-      fi
-      AC_SUBST(MOC)
-      AC_SUBST(UIC)
-      AC_SUBST(RCC)
+      #MOC=`$PKG_CONFIG --variable=moc_location QtCore`
+      #UIC=`$PKG_CONFIG --variable=uic_location QtCore`
+      #RCC=`$PKG_CONFIG --variable=rcc_location QtCore`
+      #if test -z "$RCC"; then
+      #  RCC="rcc"
+      #fi
+      #AC_SUBST(MOC)
+      #AC_SUBST(UIC)
+      #AC_SUBST(RCC)
       PKG_CHECK_MODULES(QTGL, [QtOpenGL >= 4.8 gl], [qt_pkgconfig_gl=true], [qt_pkgconfig_gl=false])
       if test "x$qt_pkgconfig_gl" = "xtrue"; then
          AC_DEFINE([HAVE_QTGL], [1], [qt has opengl support])
